daphne_worker_test/tests/e2e.rs

```diff
@@ -490,6 +490,27 @@ async fn e2e_leader_collect_abort_unknown_request() {
     assert_eq!(resp.status(), 400);
 }
 
+#[tokio::test]
+#[cfg_attr(not(feature = "test_e2e"), ignore)]
+async fn e2e_leader_collect_accept_max_batch_interval() {
+    let t = TestRunner::default().await;
+    let client = t.http_client();
+    let batch_interval = t.batch_interval();
+
+    // Maximum allowed batch duration.
+    let collect_req = CollectReq {
+        task_id: t.task_id.clone(),
+        batch_interval: Interval {
+            start: batch_interval.start - t.max_batch_duration / 2,
```

Comment:
Why subtract `t.max_batch_duration / 2`?

Reply:
This is so that `batch_interval.start` and `batch_interval.end()` both fit into the `min_batch_interval_start` and `max_batch_interval_end` limits.

Reply:
I see. It would be cleaner to construct `batch_interval` yourself, rather than use `t.batch_inteval()`, which is used as the default.

Reply:
Great idea, I've implemented this.